### PR TITLE
Copy bccsp into fabric-ca tree

### PR DESCRIPTION
**Pull request by mastersingh24**
#### Type of change

- Improvement (tree structure)

#### Description

bccsp is not actually designed to be a consumable package
outside of the fabric src tree.  We also really do not properly
maintain the dependency within fabric-ca and we plan to
pare it down to what is needed in fabric in any case.
Going forward in fabric-ca, we will also implement the
bare minimu of crypto needed.

Given all of this, we stop vendoring bccsp and simply copy
into the fabric-ca tree.  This is based on https://github.com/hyperledger/fabric/tree/e37aa35178fdb83228f63ef4670c1a67896f081b/bccsp.
This commit was chosen to maintain support for bccsp plugins which are support in 1.4.x releases but no longer supported in subsequent releases.

### CODE REVIEW

**Comment by sykesm** [comment]
Another ask: Can you please clearly indicate in the commit message (or code, script, readme, something) the revision (branch and commit) of bccsp that you chose as your base so it's easier to compare in the future? Thanks.

File: internal/pkg/bccsp/factory/test-plugin/plugin.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ package main
 import (
 	"hash"
 
-	"github.com/hyperledger/fabric/bccsp"
+	"github.com/hyperledger/fabric-ca/internal/pkg/bccsp"
```
Comment (sykesm) [inline internal/pkg/bccsp/factory/test-plugin/plugin.go:11]: This is going to be problematic for plugins.

Comment (mastersingh24) [reply internal/pkg/bccsp/factory/test-plugin/plugin.go:11]: Ah ... right ... I moved it into internal in master but it needs to be external here.

**Comment by mastersingh24** [comment]
> Without digging in, a couple of high level questions.
> 
> 1. Why are we doing this in `release-1.4` before master? We conventionally work in master and roll things back as necessary.
> 2. Do we need to pull in all of the bccsp/idemix packages? I do not believe that the CA uses them.
> 
> ```
> $ go list -f '{{ join .Deps  "\n" }}' ./...  | sort | uniq | grep bccsp
> github.com/hyperledger/fabric-ca/vendor/github.com/hyperledger/fabric/bccsp
> github.com/hyperledger/fabric-ca/vendor/github.com/hyperledger/fabric/bccsp/factory
> github.com/hyperledger/fabric-ca/vendor/github.com/hyperledger/fabric/bccsp/signer
> github.com/hyperledger/fabric-ca/vendor/github.com/hyperledger/fabric/bccsp/sw
> github.com/hyperledger/fabric-ca/vendor/github.com/hyperledger/fabric/bccsp/utils
> ```

I have a change ready for master, but it's different than the change for release-1.4 which does not **yet** suffer from the problem of being pinned to a specific commit of bccsp.

Will look into how much of the idemix can be removed in a future change ... fabric-ca does use idemix though.

File: internal/pkg/bccsp/factory/test-plugin/plugin.go (hunk 1 above)
Comment (mastersingh24) [reply internal/pkg/bccsp/factory/test-plugin/plugin.go:11]: will move this under pkg/bccsp for now

**Comment by sykesm** [comment]
> fabric-ca does use idemix though.

Yes, but does it use bccsp/idemix?

**Comment by mastersingh24** [comment]
> Another ask: Can you please clearly indicate in the commit message (or code, script, readme, something) the revision (branch and commit) of bccsp that you chose as your base so it's easier to compare in the future? Thanks.

Yeah - will add to commit as well as readme within the folder

**Comment by sykesm** [comment]
If we do #124, do we still need this?

**Comment by mastersingh24** [comment]
will not do this in this release stream

**Pull request closed without merging**